In [2]:
import numpy as np
import pandas as pd
from scipy.stats import binom
import matplotlib.pyplot as plt

In [3]:
my_data = pd.read_csv('aggregation_ops.csv')
my_data

ID        Date     First        Last                 Full GRADE LEVEL  \
0    1  10/14/2024      John       Smith           John Smith      Junior   
1    2  10/15/2024       Sam     Johnson          Sam Johnson    Freshman   
2    3  10/16/2024       Ted    Williams         Ted Williams      Senior   
3    4  10/17/2024     Simon       Brown          Simon Brown      Junior   
4    5  10/18/2024      Doug       Jones           Doug Jones    Freshman   
5    6  10/19/2024     Scott      garcia         Scott garcia   Sophomore   
6    7  10/20/2024  Jonathan      Miller      Jonathan Miller      Junior   
7    8  10/21/2024    Jeremy       Davis         Jeremy Davis    Freshman   
8    9  10/22/2024  Nickolas  Rodoriquez  Nickolas Rodoriquez      Senior   
9   10  10/23/2024    Nikita    Martinez      Nikita Martinez   Sophomore   
10  11  10/24/2024   Kaitlyn   Hernandez    Kaitlyn Hernandez   Sophomore   
11  12  10/25/2024      Mark       Lopez           Mark Lopez      Junior   
12  13  10/26/2024     James    Gonzales       James Gonzales      Junior   
13  14  10/27/2024    Nathan      Wilson        Nathan Wilson      Senior   
14  15  10/28/2024   Tiffany    Anderson     Tiffany Anderson      Junior   
15  16  10/29/2024     Sarah      Thomas         Sarah Thomas      Junior   
16  17  10/30/2024     Janet      Taylor         Janet Taylor      Junior   
17  18  10/31/2024   Samatha       Moore        Samatha Moore   Sophomore   

    STATUS  
0       OK  
1   Hungry  
2    Tired  
3   Hungry  
4       OK  
5       OK  
6       OK  
7       OK  
8       OK  
9       OK  
10  Hungry  
11      OK  
12   Tired  
13      OK  
14  Hungry  
15      OK  
16  Hungry  
17      OK

In [4]:
Part_One = my_data[ ['ID','Last' , 'STATUS','Full'] ][0:3]
Part_One

ID      Last  STATUS          Full
0   1     Smith      OK    John Smith
1   2   Johnson  Hungry   Sam Johnson
2   3  Williams   Tired  Ted Williams

In [5]:
Part_Two = my_data[ ['ID','Last', 'STATUS','Full'] ][14:17]
Part_Two

ID      Last  STATUS              Full
14  15  Anderson  Hungry  Tiffany Anderson
15  16    Thomas      OK      Sarah Thomas
16  17    Taylor  Hungry      Janet Taylor

In [8]:
Part_Three = pd.concat([Part_One,Part_Two], keys=['Part_One','Part_Two'])
Part_Three

ID      Last  STATUS              Full
Part_One 0    1     Smith      OK        John Smith
         1    2   Johnson  Hungry       Sam Johnson
         2    3  Williams   Tired      Ted Williams
Part_Two 14  15  Anderson  Hungry  Tiffany Anderson
         15  16    Thomas      OK      Sarah Thomas
         16  17    Taylor  Hungry      Janet Taylor

In [18]:
TOTAL = (Part_Three['STATUS'].count())
TOTAL

6

In [19]:
filtered1 = Part_Three[ Part_Three['STATUS'] == 'Hungry' ]
filtered1

ID      Last  STATUS              Full
Part_One 1    2   Johnson  Hungry       Sam Johnson
Part_Two 14  15  Anderson  Hungry  Tiffany Anderson
         16  17    Taylor  Hungry      Janet Taylor

In [30]:
PART1 = (filtered1['STATUS'].count())
PART1

3

In [32]:
print('Percentage of people who are hungry')
PERCENT1 = ((PART1 / TOTAL )*100)
PERCENT1

Percentage of people who are hungry


50.0

In [26]:
filtered2 = Part_Three[ Part_Three['STATUS'] == 'Tired' ]
filtered2

ID      Last STATUS          Full
Part_One 2   3  Williams  Tired  Ted Williams

In [27]:
PART2 = (filtered2['STATUS'].count())
PART2

1

In [33]:
print('Percentage of people who are Tired')
PERCENT2 = ((PART2 / TOTAL )*100)
PERCENT2

Percentage of people who are hungry


16.666666666666664

In [28]:
filtered3 = Part_Three[ Part_Three['STATUS'] == 'OK' ]
filtered3

ID    Last STATUS          Full
Part_One 0    1   Smith     OK    John Smith
Part_Two 15  16  Thomas     OK  Sarah Thomas

In [29]:
PART3 = (filtered3['STATUS'].count())
PART3

2

In [34]:
print('Percentage of people who are hungry')
PERCENT3 = ((PART3 / TOTAL )*100)
PERCENT3

Percentage of people who are hungry


33.333333333333329

In [47]:
print('Hungry') 
print(PERCENT1)
print('-----')
print('Tired') 
print(PERCENT2)
print('-----')
print('OK') 
print(PERCENT3)

Hungry
50.0
-----
Tired
16.6666666667
-----
OK
33.3333333333


Getting the formatting to work was difficult, especially with adding a % (Any formatting commands?)
From our sample size of the population we can tell that 50% are Hungrh, 33.33% are OK and 16.67% are Tired
We can extrapolate that someone should bring food to the meeting instead of coffee
Also I dont think this sample size is indicitive of the larger population, with over 50% being OK in our population